src/infrastructure/chatwoot/sync_reopen_test.go

```
@@ -0,0 +1,540 @@
+
+// fakeMediaDownloader stands in for the WhatsApp download: it writes a small
+// temp file and returns its path, which syncMessageWithOptions removes after
+// posting, so every call has to produce a fresh one.
+func fakeMediaDownloader(t *testing.T) func(context.Context, *domainChatStorage.Message, *whatsmeow.Client) (string, error) {
+	t.Helper()
+	return func(_ context.Context, msg *domainChatStorage.Message, _ *whatsmeow.Client) (string, error) {
+		f, err := os.CreateTemp(t.TempDir(), "media-*.jpg")
+		if err != nil {
+			return "", err
+		}
+		defer f.Close()
```

golangci-lint (2.13.2)

[error] Error return value of `f.Close` is not checked

(errcheck)